### PR TITLE
React QS updated to improve access token storage

### PR DESCRIPTION
**Pull request by lukeocodes**

### CODE REVIEW

File: 01-Login/src/Auth/Auth.js [hunk 1]
```diff
@@ -17,6 +20,11 @@ export default class Auth {
     this.logout = this.logout.bind(this);
     this.handleAuthentication = this.handleAuthentication.bind(this);
     this.isAuthenticated = this.isAuthenticated.bind(this);
+    this.goTo = this.goTo.bind(this);
+
+    if (localStorage.getItem('loggedIn') === 'true') {
+      this.renewAuthentication();
+    }
   }
```
Comment (aaguiarz) [inline 01-Login/src/Auth/Auth.js]: RenewAuthentication() ends up calling checkSession() which is async, so maybe it should not be called from the constructor

File: 01-Login/src/Auth/Auth.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+  accessToken;
+
   auth0 = new auth0.WebAuth({
     domain: AUTH_CONFIG.domain,
     clientID: AUTH_CONFIG.clientId,
     redirectUri: AUTH_CONFIG.callbackUrl,
     audience: `https://${AUTH_CONFIG.domain}/userinfo`,
-    responseType: 'token id_token',
+    responseType: 'token',
     scope: 'openid'
```
Comment (aaguiarz) [inline 01-Login/src/Auth/Auth.js]: We should use 'token id_token'

Also, let's remove the audience and scope, to simplify the code and introduce less concepts. Scope is 'openid profile email' by default in Auth0.js

File: 01-Login/src/Auth/Auth.js [hunk 3]
```diff
@@ -25,40 +33,52 @@ export default class Auth {
...
-    localStorage.setItem('access_token', authResult.accessToken);
-    localStorage.setItem('id_token', authResult.idToken);
-    localStorage.setItem('expires_at', expiresAt);
-    // navigate to the home route
-    history.replace('/home');
+  renewAuthentication() {
+    this.auth0.checkSession({}, (err, authResult) => {
+        if (err) {
+          this.logout();
+          console.log(err);
```
Comment (aaguiarz) [inline 01-Login/src/Auth/Auth.js]: When we call renewAuthentication() when the application starts and checkSession() fails, we should not give an error but set the application to 'unauthenticated' state. Check how it's done in the Angular sample that Dan did.

File: 02-Calling-an-API/src/Auth/Auth.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  goTo(path) {
+    history.replace(path);
+  }
+
+  getAccessToken() {
+    if (!this.isTokenValid()) {
+      this.renewAuthentication();
+    }
```
Comment (aaguiarz) [inline 02-Calling-an-API/src/Auth/Auth.js]: To get the access token we need to hook on the callback of checkSession(), here we are not waiting until the checkSession() flow finishes. This implies getAccessToken() should call a callback when finished, and it should be called with a callback function.

File: 02-Calling-an-API/src/Auth/Auth.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        this.logout();
+      } else if (authResult) {
+        this.logUserIn(authResult);
+      }
+    });
+  }
+
+  logUserIn(authResult) {
+    this.expires = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
+    this.accessToken = authResult.accessToken;
```
Comment (aaguiarz) [inline 02-Calling-an-API/src/Auth/Auth.js]: we don't need to stringify the date. It was done because we were storing it in localStorage

**Comment by lukeocodes** [comment]
Closed PR but I'll leave the branch/fork around.

**Pull request closed without merging**